scala.wake

```diff
@@ -443,38 +443,43 @@ target memoizeMillCompile name \ module =
 
   def transDeps = module.getScalaModuleTransDeps
   def buildFile = millBuildFile name module
-  def inputs =
+  def inputsResult =
     def sourceFiles = module.getScalaModuleAllSourceFiles ++ module.getScalaModuleResourceFiles
-    def depJars = map compileScalaModule transDeps
-    millBin, buildFile, depJars ++ sourceFiles ++ (ivyCacheDeps Unit)
-    | sortBy (_.getPathName <* _.getPathName) # Sorted to ensure hash is stable
-  # This is a hack to ensure that changes to visible files rerun the job
-  # We give directories as inputs to Bloop which then looks in those directories so new files m
-  def visibleHashHack =
-    def hash = (\x prim "hash") inputs
-    "VISIBLE_FILES_HASH_HACK={format hash}"
-  def cmd =
-    "bash", relMillBin, "--no-default-predef", "--home", relMillHome, "show", "{name}.jar", Nil
-  def plan =
-    makePlan cmd inputs
-    | editPlanEnvironment ("COURSIER_CACHE={relIvyCache}", visibleHashHack, _)
-    | setPlanLabel "Compiling {name}..."
-    | setPlanDirectory buildDir
-    # Mill uses the entire classpath for caching so it must be local only for incremental compile
-    | setPlanLocalOnly True
-    | setPlanFnOutputs (\_ jarName, Nil)
-    | setPlanResources resources
-    | setPlanStdout logVerbose
-    | setPlanStderr logVerbose
-  def echoOnError job path = match path.getPathError
-    None   = path
-    Some e =
-      def _ = print (formatError job plan)
-      path
-  def job = plan | runJob
-  job
-  | getJobOutput
-  | echoOnError job
+    def depJarsResult = transDeps | map compileScalaModule | map getPathResult | findFail
+    def addOtherInputs depJars =
+      millBin, buildFile, depJars ++ sourceFiles ++ (ivyCacheDeps Unit)
+      | sortBy (_.getPathName <* _.getPathName) # Sorted to ensure hash is stable
+    depJarsResult
+    | rmap addOtherInputs
+  match inputsResult
+    Pass inputs =
+      # This is a hack to ensure that changes to visible files rerun the job
+      # We give directories as inputs to Bloop which then looks in those directories so new files m
+      def visibleHashHack =
+        def hash = (\x prim "hash") inputs
+        "VISIBLE_FILES_HASH_HACK={format hash}"
+      def cmd =
+        "bash", relMillBin, "--no-default-predef", "--home", relMillHome, "show", "{name}.jar", Nil
+      def plan =
+        makePlan cmd inputs
+        | editPlanEnvironment ("COURSIER_CACHE={relIvyCache}", visibleHashHack, _)
+        | setPlanDirectory buildDir
+        # Mill uses the entire classpath for caching so it must be local only for incremental compile
+        | setPlanLocalOnly True
```

Comment:
Does this still need to be `LocalOnly`?

Reply:
Oh I just noticed the comment above, can you expand a little on this, does it mean that the whole classpath should be in the visible files list?

Reply:
Yeah, and it's good you asked because I forgot and had to remind myself which means it needs to be in the comment:

mill uses the filepaths in determining file identity, so `.fuse` shows up if it's not `LocalOnly`. With alternative sandboxing, it would not need to be `LocalOnly`.

Reply:
`.fuse/pid` shouldn't be visible anymore with wake 0.19+ as we now mount the .fuse/pid directory over the top of the workspaces

See: https://github.com/sifive/wake/pull/444/files#diff-b61ba27252888d488d262f54042f9357R188

Reply:
👍 on not making it LocalOnly.